pkg/nordic_softdevice_ble/src/gnrc_nordic_ble_6lowpan.c

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (c) 2016, Nordic Semiconductor
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the Institute nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ */
+
+/**
+ * @{
+ * @ingroup     net
+ * @file
+ * @brief       Glue for Nordic's SoftDevice BLE 6lowpan blob to netapi
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @}
+ */
+
+#include <assert.h>
+#include <errno.h>
+
+#include "msg.h"
+#include "thread.h"
+
+#include "net/gnrc.h"
+#include "net/gnrc/nettype.h"
+
+#include "ble-core.h"
+#include "softdevice_handler.h"
+
+#include "ble-mac.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#if defined(MODULE_OD) && ENABLE_DEBUG
+#include "od.h"
+#endif
+
+#define BLE_NETAPI_MSG_QUEUE_SIZE   (8U)
+#define BLE_PRIO                    (THREAD_PRIORITY_MAIN - 1)
+
+kernel_pid_t gnrc_nordic_ble_6lowpan_pid;
+static char _stack[(THREAD_STACKSIZE_DEFAULT + DEBUG_EXTRA_STACKSIZE)];
+
+static uint8_t _own_mac_addr[BLE_SIXLOWPAN_L2_ADDR_LEN];
+
+static uint8_t _sendbuf[BLE_SIXLOWPAN_MTU];
+
+static void _ble_mac_callback(ble_mac_event_enum_t event, void* arg)
+{
+    msg_t m = { .type=event, .content.ptr=arg };
+
+    if (!msg_send_int(&m, gnrc_nordic_ble_6lowpan_pid)) {
+        puts("_ble_mac_callback(): possibly lost interrupt");
+    }
+}
+
+static void _handle_raw_sixlowpan(ble_mac_inbuf_t *inbuf)
```

Comment:
This name is somewhat misleading IMHO. I first thought you are handling uncompressed IPv6 packets here, ignoring `gnrc_sixlowpan`.